src/core/sign.ml

```diff
@@ -15,17 +15,22 @@ type inductive =
   { ind_cons  : sym list  (** List of constructors                 *)
   ; ind_prop  : sym       (** Induction principle on propositions. *) }
 
+(** Syntactic sugar properties of symbols. They are linked to symbols to
+    provide syntax extensions to these symbols. *)
+type synt_sugar =
+  | Prefix of unop (** Prefix (or unary) operator. *)
+  | Infix of binop (** Infix (or binary) operator. *)
```

Comment:
What about replacing unop by priority, and binop by priority*associativity, as you don't need the other elements of unop and binop that are useful for handling the set prefix/infix commands only?

Reply:
We may need the string for printing purposes, don't we? Otherwise we won't remember, for instance, which is the associated infix representation of `plus` when we have `set infix "+" := plus`.

Reply:
And do you think we should propagate those modifications on the `pp_hint` of `Sig_state`? I must admit that all the `pp_hint` functions are not really clear to me.

Reply:
First question: right but is it a good idea to keep the set infix/prefix commands as they currently are? Until now, these commands had two functions: 1) declare a non-regular identifier like set declared, 2) map this non-regular identifier to a (qualified) regular one. The function 1) is useless now. So, I think that we should also change the set infix/prefix commands. Instead of having: set infix/prefix ... string_lit := qident, we should have: set infix/prefix ... qident, the meaning of which would simply be add a new notation for qident. What do you think?

Second question: yes.

Reply:
Ah yes. Could we leave this as a later pull request, provided that we create an issue for the record?

Reply:
Ok, I can take care of these two items in a next PR. Just, could you please remove the type notation, move the type pp_hint to sign.ml, and take sign_syntax:pp_hint SymMap.t ref?